### PR TITLE
[instrument_list] Remove superglobal access

### PR DESCRIPTION
**Pull request by driusan**
This refactors the fix in #3721 in order to remove references
to superglobals. Instead, the variables are instantiated on the
module level from the PSR object so that the page can assume
they exist.

### CODE REVIEW

**Comment by driusan** [comment]
I'm sending this to 20.0-release because that's the only place that #3721 exists, but it can go into a release after that.

**Comment by johnsaigle** [comment]
@driusan I rebased for you

File: modules/instrument_list/php/instrument_list.class.inc [hunk 1]
```diff
@@ -101,33 +125,13 @@ class Instrument_List extends \NDB_Menu_Filter
...
-        $gets = $request->getQueryParams();
-
-        $attribute = $request->getAttribute("CandID");
-        if ($attribute !== null) {
-            $this->candidate = \Candidate::singleton($attribute);
-        } else {
-            $this->candidate = \Candidate::singleton($gets['candID']);
+        $this->candidate = $request->getAttribute("Candidate");
+        $this->timePoint = $request->getAttribute("timePoint");
+        if ($this->candidate == null || $this->timePoint === null) {
```
Comment (PapillonMcGill) [inline modules/instrument_list/php/instrument_list.class.inc]: Is there a reason why for candidate `==` is used while `===` is used for timepoint? Would be better if both use `===`

File: modules/instrument_list/php/instrument_list.class.inc [hunk 2]
```diff
@@ -47,6 +48,29 @@ class Instrument_List extends \NDB_Menu_Filter
...
+     *
+     * @return void
+     */
+    public function setTimePoint(\TimePoint $timepoint)
+    {
+        $this->timePoint = $timepoint;
+    }
+
+    /**
+     * Sets the candidate is being accessed on this page
```
Comment (zaliqarosli) [inline modules/instrument_list/php/instrument_list.class.inc]: Sets the candidate that is being accessed ..

File: modules/instrument_list/php/instrument_list.class.inc (hunk 1 above)
Comment (zaliqarosli) [reply modules/instrument_list/php/instrument_list.class.inc]: i agree. this seems to be a typo

**Comment by driusan** [comment]
@zaliqarosli / @PapillonMcGill should be fixed, if there was a reason I can't remember why one was using double equals Travis should catch it..

**Pull request merged**